### DISCUSSION TITLE
osqueryd output to CSV

### DISCUSSION THREAD

**Discussion by Hemro** [Security-Onion-Solutions/securityonion#4226] [Unsupported Versions] [answered]
Can you configure output of osqueryd to CSV, I know you can do it with osqueryi

**Comment by defensivedepth** [comment]
Can you give some further context around what you are trying to do?

osqueryd & osqueryi are actually the same binary - 

![image](https://user-images.githubusercontent.com/954732/118860042-eae7ac00-b8a8-11eb-81f2-7ccfad9cd84d.png)

**Comment by Hemro** [comment]
I am trying to automate a scheduled snapshot output from fleet to csv.

**Reply by defensivedepth** [reply]
The scheduled query results outputted by FleetDM cannot be output in CSV format; However, Logstash does have a CSV output: https://www.elastic.co/guide/en/logstash/current/plugins-outputs-csv.html

Can you give me further context with what you want to do with the CSV?

**Comment by Hemro** [comment, marked as answer]
The generated csv file will be imported into a currently beta application. e.g AD users

**Reply by defensivedepth** [reply]
You will need to figure out something custom. The Fleet scheduled query results are in JSON in `/nsm/osquery/fleet/result.log`

You could try scheduling a script to convert that json to csv every so often.... or use the Logstash to CSV output plugin etc.
